x/concentrated-liquidity/incentives.go

```diff
@@ -0,0 +1,51 @@
+package concentrated_liquidity
+
+import (
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/osmoutils/accum"
+	"github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity/types"
+)
+
+const (
+	uptimeAccumPrefix = "uptime"
+)
+
+// createUptimeAccumulators creates accumulator objects in store for each supported uptime for the given poolId.
+// The accumulators are initialized with the default (zero) values.
+func (k Keeper) createUptimeAccumulators(ctx sdk.Context, poolId uint64) error {
+	for uptimeId := range types.SupportedUptimes {
+		err := accum.MakeAccumulator(ctx.KVStore(k.storeKey), getUptimeAccumulatorName(poolId, uint64(uptimeId)))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getUptimeAccumulatorName(poolId uint64, uptimeId uint64) string {
+	poolIdStr := strconv.FormatUint(poolId, uintBase)
+	uptimeIdStr := strconv.FormatUint(uptimeId, uintBase)
+	return strings.Join([]string{uptimeAccumPrefix, uptimeIdStr, poolIdStr}, "/")
+}
+
+// nolint: unused
+// getUptimeAccumulators gets the uptime accumulator objects for the given poolId
+// Returns error if accumulator for the given poolId does not exist.
+func (k Keeper) GetUptimeAccumulators(ctx sdk.Context, poolId uint64) ([]accum.AccumulatorObject, error) {
```

Comment:
Does this have to be exported?

Reply:
I figured it could be useful & it's a getter which we tend to make public in GAMM

Reply:
Ended up making private and exporting